src/utils/StreamHighlightUtils.ts

```diff
@@ -77,13 +77,21 @@ export class StreamHighlightUtils {
 
 export function getRestHighlightsForAllTerms(
   toHighlight: string,
-  termsToHighlight: { [originalTerm: string]: string[] },
-  phrasesToHighlight: { [phrase: string]: { [originalTerm: string]: string[] } },
+  termsToHighlight: IHighlightTerm,
+  phrasesToHighlight: IHighlightPhrase,
   opts: IStreamHighlightOptions
 ): IHighlight[] {
   const indexes = [];
   const sortedTerms = _.keys(termsToHighlight).sort(termsSorting);
-  _.each(sortedTerms, (term: string) => {
+
+  const phrasesTerms = _.chain(phrasesToHighlight)
+    .values()
+    .map(_.keys)
+    .flatten()
+    .value();
+  const uniqueTerms = _.difference(sortedTerms, phrasesTerms);
```

Comment:
What is the difference between `sortedTerms` and `phrasesTerms`? I ask because here you made the decision to exclude duplicates from being rendered as `sortedTerms`. Is there a reason why we would maybe exclude it from the `phrasesTerms` instead? Or does it not matter?

Reply:
After checking, only this way is possible, as the algorithm doesn't use the phrases terms directly, only the phrases. E.g.:
```
const phrasesToHighlight: IHighlightPhrase = { 'test phrase': { phrase: [], test: [] } };
```
It will split the `test phrase`, not use the terms, so excluding the terms wouldn't work

Reply:
Sorted terms come from `termsToHighlight` sorted by length
Phrases terms come from `phrasesToHighlight` own terms